crates/indexer/src/geyser/accounts/spl_governance.rs

```diff
@@ -0,0 +1,713 @@
+        governing_token_owner: Owned(data.governing_token_owner.to_string()),
+        governing_token_deposit_amount: data.governing_token_deposit_amount.try_into()?,
+        unrelinquished_votes_count: data.unrelinquished_votes_count.try_into()?,
+        total_votes_count: data.total_votes_count.try_into()?,
+        outstanding_proposal_count: data.outstanding_proposal_count.try_into()?,
+        reserved: Owned(data.reserved.to_vec()),
+        governance_delegate: data.governance_delegate.map(|d| Owned(d.to_string())),
+        slot: slot.try_into()?,
+        write_version: write_version.try_into()?,
+    };
+
+    client
+        .db()
+        .run(move |db| {
+            insert_into(token_owner_records_v2::table)
+                .values(&row)
+                .on_conflict(token_owner_records_v2::address)
+                .do_update()
+                .set(&row)
+                .execute(db)
+        })
+        .await
+        .context("Failed to insert vote record v2")?;
+
+    Ok(())
+}
+
+pub(crate) async fn process_signatory_record_v2(
+    client: &Client,
+    key: Pubkey,
+    data: SignatoryRecordV2,
+    slot: u64,
+    write_version: u64,
+) -> Result<()> {
+    let row = DbSignatoryRecordV2 {
+        address: Owned(key.to_string()),
+        account_type: match data.account_type {
+            GovernanceAccountType::SignatoryRecordV2 => {
+                GovernanceAccountTypeEnum::SignatoryRecordV2
+            },
+            GovernanceAccountType::SignatoryRecordV1 => {
+                GovernanceAccountTypeEnum::SignatoryRecordV1
+            },
+            _ => GovernanceAccountTypeEnum::Uninitialized,
+        },
+        proposal: Owned(data.proposal.to_string()),
+        signatory: Owned(data.signatory.to_string()),
+        signed_off: data.signed_off,
+        slot: slot.try_into()?,
+        write_version: write_version.try_into()?,
+    };
+
+    client
+        .db()
+        .run(move |db| {
+            insert_into(signatory_records_v2::table)
+                .values(&row)
+                .on_conflict(signatory_records_v2::address)
+                .do_update()
+                .set(&row)
+                .execute(db)
+        })
+        .await
+        .context("Failed to insert signatory record v2")?;
+
+    Ok(())
+}
+
+#[allow(clippy::too_many_lines)]
+pub(crate) async fn process_proposal_v2(
+    client: &Client,
+    key: Pubkey,
+    data: ProposalV2,
+    slot: u64,
+    write_version: u64,
+) -> Result<()> {
+    let (vote_threshold_type, vote_threshold_percentage) =
+        if let Some(VoteThresholdPercentage::YesVote(p)) = data.vote_threshold_percentage {
+            (Some(VoteThresholdEnum::YesVote), Some(i16::try_from(p)?))
+        } else if let Some(VoteThresholdPercentage::Quorum(p)) = data.vote_threshold_percentage {
+            (Some(VoteThresholdEnum::Quorum), Some(i16::try_from(p)?))
+        } else {
+            (None, None)
+        };
```

Comment:
This seems like a good item for the Relms team to review, yeah?

Reply:
> This seems like a good item for the Relms team to review, yeah?

Its an Option<VoteThresholdPercentage> which I have broken into two columns of a table.
https://docs.rs/spl-governance/2.1.1/spl_governance/state/enums/enum.VoteThresholdPercentage.html